packages/plugin-kit/src/fs/readSliceTemplateLibrary.ts

```diff
@@ -0,0 +1,77 @@
+import fs from "node:fs/promises";
+import path from "node:path";
+
+import { checkIsTypeScriptProject } from "./checkIsTypeScriptProject";
+import { SliceMachineHelpers } from "../createSliceMachineHelpers";
+import { SharedSliceContent } from "@prismicio/types-internal/lib/content";
+import { SharedSlice } from "@prismicio/types-internal/lib/customtypes";
+import { SliceTemplateLibraryReadHookReturnType } from "../hooks/sliceTemplateLibrary-read";
+
+export type ReadSliceTemplateLibraryArgs = {
+	helpers: SliceMachineHelpers;
+	dirName: string;
+	templateIDs?: string[];
+	templates: {
+		mocks: SharedSliceContent[];
+		screenshotPaths: Record<string, string>;
+		model: SharedSlice;
+	}[];
+	componentFileNames: {
+		js: string;
+		ts: string;
+	};
+};
+
+export const readSliceTemplateLibrary = async (
+	args: ReadSliceTemplateLibraryArgs,
+): Promise<SliceTemplateLibraryReadHookReturnType> => {
+	const isTypeScriptProject = await checkIsTypeScriptProject(args);
+
+	const {
+		templateIDs,
+		templates: initialTemplates,
+		dirName,
+		componentFileNames,
+	} = args;
+
+	const templates =
+		templateIDs && templateIDs.length
+			? initialTemplates.filter((t) => templateIDs?.includes(t.model.id))
+			: initialTemplates;
+
+	const templatesPromises = templates.map(async (t) => {
+		const { mocks, model, screenshotPaths } = t;
+
+		const screenshotEntries = Object.entries(screenshotPaths);
+		const screenshotPromises = screenshotEntries.map(([key, filePath]) => {
+			return fs
+				.readFile(path.join(dirName, filePath))
+				.then((data) => [key, data]);
+		});
+		const readScreenshots = await Promise.all(screenshotPromises);
+		const screenshots = Object.fromEntries(readScreenshots);
+
+		const fileName = isTypeScriptProject
+			? componentFileNames.ts
+			: componentFileNames.js;
+
+		// import fileNames object from templates if ever needed
```

Comment:
Can you explain me this comment, not sure to get it?

Reply:
Actually did it :/